frontend/src/components/ExpenseList.js

```diff
@@ -22,26 +23,60 @@ class ExpenseList extends React.Component {
       { categoryid: 11, categoryName: "Miscellenous" },
     ],
   };
-  render() {
 
+  onChange = (e) => {
+    // this changes the state of userexpenses amount.
+    this.setState((prevState) => {
+      let userexpense = Object.assign({}, prevState.userexpense);
+      userexpense[e.target.name] = e.target.value;
+      return { userexpense };
+    });
+  };
+
+  onSubmit = (e) => {
+    e.preventDefault();
+    this.props.addExpense(
+      this.state.userexpense.categoryName,
+      this.state.userexpense.amount
+    );
+    this.setState({ categoryName: "", amount: "" });
+    e.target.reset();
+  };
+
+  render() {
     const categoriesList = this.state.categories.map((item) => {
       return <option>{item.categoryName}</option>;
     });
-    const { saveButtonLabel } = require("../phrases/App.json");
-
 
     return (
       <>
-        <Form className="expenseform">
+        <Form onSubmit={this.onSubmit.bind(this)}>
           <Form.Group controlId="expenseList" className="expenseform">
-            <Form.Control as="select" custom defaultValue="Add category">
+            <Form.Control
+              as="select"
+              custom
+              defaultValue="Add category"
+              name="categoryName"
+              onChange={this.onChange.bind(this)}
+              required
+            >
               <option disabled>Add category</option>
```

Comment:
Change this to fix the empty category being accepted:
```jsx
<option value="">Add category</option>
```

Reply:
This is working good. Thanks for the fix...!!!